### PR TITLE
Moving online install into go to support automation

### PR DESCRIPTION
**Pull request by marccampbell**
This is related to replicatedhq/kots#450

### CODE REVIEW

File: pkg/app/app.go [hunk 1]
```diff
@@ -66,6 +70,86 @@ func GetFromSlug(slug string) (*App, error) {
...
+	}
+
+	installState := ""
+	if strings.HasPrefix(upstreamURI, "replicated://") == false {
+		installState = "installed"
+	} else {
+		if isAirgapEnabled {
+			installState = "airgap_upload_pending"
+		} else {
+			installState = "online_upload_pending"
```
Comment (divolgin) [inline pkg/app/app.go:133]: wouldn't `replicated://` fall into this  category? this might need a comment.

Comment (marccampbell) [reply pkg/app/app.go:133]: Yeah, the if else statement covers this?

File: web/src/components/UploadAirgapBundle.jsx [hunk 2]
```diff
@@ -216,23 +215,36 @@ class UploadAirgapBundle extends React.Component {
...
+      }),
     })
-    .catch((err) => {
-      err.graphQLErrors.map(({ msg }) => {
-        this.setState({
-          preparingOnlineInstall: false,
-          onlineInstallErrorMessage: msg
-        });
+    .then(async (result) => {
+      const body = await result.json();
```
Comment (divolgin) [inline web/src/components/UploadAirgapBundle.jsx:236]: missing error handling?

File: web/src/components/UploadLicenseFile.jsx [hunk 3]
```diff
@@ -61,16 +53,33 @@ class UploadLicenseFile extends React.Component {
...
+      headers: {
+        "Authorization": Utilities.getToken(),
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        licenseData: licenseText,
+      }),
+    })
+    .then(async (result) => {
+      const body = await result.json();
```
Comment (divolgin) [inline web/src/components/UploadLicenseFile.jsx:73]: no error handling

**Pull request merged**